pkg/operation/botanist/dnsrecord.go

```diff
@@ -206,3 +211,38 @@ func (b *Botanist) deployOrRestoreDNSRecord(ctx context.Context, dnsRecord compo
 	}
 	return dnsRecord.Deploy(ctx)
 }
+
+// CleanupOrphanedDNSRecordSecrets cleans up secrets related to DNSRecords which may be orphaned after introducing the 'dnsrecord-' prefix
+func (b *Botanist) CleanupOrphanedDNSRecordSecrets(ctx context.Context) error {
+	internalDNSRecordSecretName := b.Shoot.Components.Extensions.InternalDNSRecord.GetSecret().Name
+	ownerDNSRecordSecretName := b.Shoot.Components.Extensions.OwnerDNSRecord.GetSecret().Name
```

Comment:
As mentioned above, getting the secret names in this way doesn't guarantee the so named secrets are actually already deployed. I believe this logic and the ifs below could simply be skipped in favor of indiscriminately deleting all secrets with old names (if they exist) which would be safe if executed at the right point in the flow, that is after all DNSRecords have been updated, e.g.:

```go
b.cleanupOrphanedDNSRecordSecret(ctx, b.Shoot.GetInfo().Name + "-" + DNSExternalName)
b.cleanupOrphanedDNSRecordSecret(ctx, b.Shoot.GetInfo().Name + "-" + DNSInternalName)
```

(with proper error handling of course)

Reply:
Thanks, done.